dtl-lexer/src/variable.rs

```diff
@@ -66,22 +67,19 @@ impl<'t> TemplateContent<'t> for FilterToken {
 }
 
 #[derive(Debug, PartialEq)]
-pub enum VariableTokenType {
-    Variable,
+pub enum VariableToken {
+    Variable(Variable),
     Int(BigInt),
     Float(f64),
 }
 
-#[derive(Debug, PartialEq)]
-pub struct VariableToken {
-    pub at: (usize, usize),
-    pub token_type: VariableTokenType,
-}
-
 impl<'t> TemplateContent<'t> for VariableToken {
     fn content(&self, template: TemplateString<'t>) -> &'t str {
-        let (start, len) = self.at;
-        &template.0[start..start + len]
+        match self {
+            VariableToken::Variable(var) => template.content(var.at),
+            VariableToken::Int(num) => num.to_string().as_str(),
+            VariableToken::Float(num) => num.to_string().as_str(),
+        }
```

Comment:
@LilyFirefly So I integrated your review and merged `VariableTokenType` and `VariableToken` but I can't solve this one. I can't find how to produce a string with lifetime in Rust. On the other hand, I think this is used only in tests so I should be able to drop it.